src/components/WorkOrder/AssetTypeField.js

```diff
@@ -0,0 +1,321 @@
+import React, { Component } from "react";
+import Autocomplete from "react-autocomplete";
+
+import { FIELDS, ASSET_TYPE_OPTIONS } from "./formConfig";
+import { getAllAssets } from "./helpers";
+
+export default class AssetTypeField extends Component {
+  constructor(props) {
+    super(props);
+
+    this.setInitalAssetName = field => {
+      const fieldArray = this.props.data[`${FIELDS.ASSETS[field].fieldId}_raw`];
+      const hasValue = fieldArray && fieldArray.length > 0;
+
+      if (hasValue) {
+        return fieldArray[0].identifier;
+      } else {
+        return "";
+      }
+    };
+
+    this.state = {
+      signalOptions: [],
+      schoolBeaconOptions: [],
+      cameraOptions: [],
+      hazardFlasherOptions: [],
+      dmsOptions: [],
+      sensorOptions: [],
+      signal: this.setInitalAssetName("signal"),
+      camera: this.setInitalAssetName("camera"),
+      schoolBeacon: this.setInitalAssetName("schoolBeacon"),
+      hazardFlasher: this.setInitalAssetName("hazardFlasher"),
+      dms: this.setInitalAssetName("dms"),
+      sensor: this.setInitalAssetName("sensor"),
+      updatedFormData: {}
+    };
+
+    this.menuStyle = {
+      borderRadius: "3px",
+      boxShadow: "0 2px 12px rgba(0, 0, 0, 0.1)",
+      background: "rgba(255, 255, 255, 0.9)",
+      padding: "2px 0",
+      fontSize: "120%",
+      position: "fixed",
+      overflow: "auto",
+      zIndex: "999",
+      maxHeight: "50%"
+    };
+
+    this.renderItem = (item, isHighlighted) => (
+      <div
+        key={item.id}
+        style={{
+          background: isHighlighted ? "lightgray" : "white",
+          padding: "2px 5px"
+        }}
+      >
+        {item.identifier}
+      </div>
+    );
+
+    this.wrapperStyle = { display: "block" };
+
+    this.shouldItemRender = (item, value) =>
+      item.identifier.toLowerCase().indexOf(value.toLowerCase()) > -1;
+
+    this.inputProps = field => {
+      return {
+        className: "form-control",
+        name: FIELDS.ASSETS[field].fieldId,
+        placeholder: "Type to search..."
+      };
+    };
+  }
+
+  componentDidMount() {
+    // It might not be ideal UX for us to wait for all 6 assets type options to return
+    // before providing any options to the user. Perhaps it would be better if we first
+    // queried the current asset type (ex: signals), set those options to state,
+    // and then query the rest. 🤔
```

Comment:
considering that 90% of the work orders will be at signals, i think this is a good idea.